system_tests/program_test.go

```diff
@@ -389,10 +389,17 @@ func storageTest(t *testing.T, jit bool) {
 	key := testhelpers.RandomHash()
 	value := testhelpers.RandomHash()
 	tx := l2info.PrepareTxTo("Owner", &programAddress, l2info.TransferGas, nil, argsForStorageWrite(key, value))
-	ensure(tx, l2client.SendTransaction(ctx, tx))
+	receipt := ensure(tx, l2client.SendTransaction(ctx, tx))
+
 	assertStorageAt(t, ctx, l2client, programAddress, key, value)
 
 	validateBlocks(t, 2, jit, builder)
+	// Without this sleep, this test fails in the "-race" CI builds complaining that
+	// the block has not been seen in L1 yet.
+	time.Sleep(10 * time.Second)
```

Comment:
@PlasmaPower or @tsahee, I'm sure there is a way for me to programmatically wait for an even from the L1 or something to ensure that the transaction is in L1, but I would love a pointer.

As the comment says, before this, there was an error being emitted:
```
[failed to get validation inputs 5 failed calculating position for validation: batch not found on L1 yet]
```

Reply:
There are two ways I can think of:
1) wait for L2 safe or finalized block to be >= blockNumber. Not sure if the test l1 supports safe/finalized - but if it does that will work. Might require submitting transactions to l1 to make sure safe/finalized advances.
2) read InboxTracker.GetBatchCount() and then GetBatchMessageCount(batch) to see the last message that's in a batch

Reply:
I went with the second option.